index.d.ts

```diff
@@ -1,6 +1,7 @@
 import { ClientRequest, IncomingMessage, ServerResponse } from "http";
 import * as opentracing from "opentracing";
 import { SpanOptions } from "opentracing/lib/tracer";
+import { ExecutionArgs, ExecutionResult } from "graphql"
```

Comment:
This doesn't work since the tracer doesn't depend on `graphql`. While unfortunate, I think the only solution is to copy the definitions.

Reply:
I thought it was enough to have `graphql` in the dev dependencies to use the type definitions: https://github.com/DataDog/dd-trace-js/blob/master/package.json#L99.

Reply:
Ahh, I see now. The version of `graphql` in the dev dependencies is relatively old. That version doesn't export any TS definitions. Would it make sense to update the dependency?

Reply:
Copying over the types would be a relatively large effort as well. Skimming over the definition files of `graphql` I can see it would require to copy over the following files/fragments of files:
- https://github.com/graphql/graphql-js/blob/157edcb4f7dfe9cb2acd3d82831430d983bbfbcf/src/type/schema.d.ts
- https://github.com/graphql/graphql-js/blob/157edcb4f7dfe9cb2acd3d82831430d983bbfbcf/src/language/ast.d.ts#L201-L297
- parts of https://github.com/graphql/graphql-js/blob/157edcb4f7dfe9cb2acd3d82831430d983bbfbcf/src/type/definition.d.ts
- https://github.com/graphql/graphql-js/blob/157edcb4f7dfe9cb2acd3d82831430d983bbfbcf/src/error/GraphQLError.d.ts
- https://github.com/graphql/graphql-js/blob/157edcb4f7dfe9cb2acd3d82831430d983bbfbcf/src/tsutils/Maybe.d.ts

The more I'm looking at it, the less convinced I am the copy approach is worth it (mostly due to maintenance implications) 😞

Reply:
The main issue is that the definitions are exposed outside the library, so devDependencies would be missing and the definitions would not work even if it's up-to-date in development.

If it's too much work to copy the definitions, I'd say to simply use `any` for now. This is a bit of an unfortunate limitation of TypeScript which pretty much requires all libraries to be monorepos exposing types as a separate package to work properly.

Reply:
I've changed it to use `any` instead: https://github.com/DataDog/dd-trace-js/pull/909/commits/a1554b505e55c7d5f0f02b94f87790352a8e5e54.